src/content/docs/telemetry-data-platform/ingest-apis/metric-api/NRQL-high-cardinality-metrics.mdx

```diff
@@ -0,0 +1,146 @@
+```
+
+Finally, if you believe you have hit one of your cardinality limits, you can confirm this by checking for a related `NrIntegrationError`:
+
+```
+FROM NrIntegrationError SELECT count(*) where name = 'CardinalityViolationException' and newRelicFeature = 'Metrics' facet cardinalityLimitType, metricName, message since today 
+```
+
+### Finding cardinality contributors: dimensions
+
+Once you've determined a metric you want to explore, the next step is to determine which dimensions in a given metric contribute the most to its cardinality. If you are unfamiliar with the values of your dimensions, you can look at them like so:
+
+```
+FROM Metric SELECT dimensions() WHERE metricName = 'memory.heap' SINCE today RAW
+```
+
+The JSON results view will likely be advisable here. Looking through these could reveal some dimensions containing a unique ID or other highly variable value that might be worth removing. 
+
+If you are already familiar with what values your attributes can take on, the `keySet()` results may be easier to scan:
+
+```
+FROM Metric SELECT keySet() WHERE metricName = 'memory.heap' SINCE today RAW
+```
+
+Understanding the dimensions that have the most influence on your total cardinality comes down to understanding how each key's values correlate with one another. You can experiment with what your cardinality would be without a dimension simply by adding it to the exclude list:
+
+```
+FROM Metric SELECT cardinality(memory.heap, exclude: {'container.id'}) SINCE today RAW
+```
+
+Likewise, there is an include list if that is more convenient to the query context:
+
+```
+FROM Metric SELECT cardinality(memory.heap, include: {'host.name', 'region'}) SINCE today RAW
+```
+
+Managing cardinality can be tricky to conceptualize, but the above methods will help you get answers to questions like "What metric is contributing the most cardinality?" and "What impact does a given attribute(s) have to that total cardinality?". 
+
+It's often the case that cardinality tracks with the most unique value, as that value may pin down the possible values other attributes can take on. However, there are plenty of cases where the explosion of possible combinations of a handful of attributes drives the total cardinality. Things that look like unique identifiers are generally a good place to start, but sometimes it's no single key but the combination of two or more keys. The more familiar you are with your data and the systems that generate it, the easier it will be to know which attributes to include or exclude. 
+
+<Callout variant="tip">
+  If you'd like to learn more about limits and troubleshooting  the Metric API, [Metric API limits and restricted attributes](/docs/telemetry-data-platform/ingest-apis/metric-api/metric-api-limits-restricted-attributes/) and [Troubleshoot Metric API with NRIntegrationError events](/docs/telemetry-data-platform/ingest-apis/metric-api/troubleshoot-nrintegrationerror-events/) are both great resources to explore. 
```

Comment:
This was difficult for me to parse. How about putting those two links in a bullet list within the tip?

If you'd like to learn more about limits and troubleshooting the Metric API, here are two good resources:

*  [Metric API limits and restricted attributes](/docs/telemetry-data-platform/ingest-apis/metric-api/metric-api-limits-restricted-attributes/) 
* [Troubleshoot Metric API with NRIntegrationError events](/docs/telemetry-data-platform/ingest-apis/metric-api/troubleshoot-nrintegrationerror-events/)

Reply:
Done!